### ISSUE TITLE
[desktop] 'Downloaded' size of a file doesn't match with the 'Size' of the file 

### ISSUE THREAD

**Issue by muckachina** [frostwire/frostwire#383]
Tested on OS: **Mac SIERRA & Yosemite, Windows 10 
FrostWire version 6.4.5 (build 222)** 
Screen: **Transfer list **
Test: **Download a file only once and check the downloaded-size of the file does not exceed the real size of the file or doesn't say 0.0KB If the file says 'Finished' or Seeding**
**Ref. S-Trans-29**

I can only reproduce this on Windows devices for Torrents and is part of the 'mess-Transfer' since last FrostWire 6.4.3 where sometimes prior downloads does not appear on the Transfer list unless a new torrent start downloading, some files appear 'Seeding' although the status is 'Not Seeding' (https://github.com/frostwire/qa-desktop/issues/102)...

**Escenario A - To reproduce on Windows**:

1) Download a file and make sure the status says 'Finished' or 'Seeding'

2) Exit FrostWire and launch the app again

3) Check all files on the Transfer list and **match the 'Size' column with the 'Downloaded' column to make sure the 'Downloaded' size does not exceed the real Size of the file or in case is a file 'Downloaded' already the size of 'Downloaded' column does not say '0.0KB'**.

![screen shot 2017-03-06 at 2 39 14 pm](https://cloud.githubusercontent.com/assets/1534723/23626509/24eaecac-027b-11e7-8f83-7ba8bed62a6e.png)

**Escenario B - On Mac and Windows affects Cloud files** 

**To reproduce Files that ends with 'Downloaded Size = 0.0.KB:**

1) Make sure the Status on Settings is 'Seeding' (Check the bottom bar)

2) Start downloading a file from YT (audio or Video)

3) When the download is complete and the file changes to 'Seeding' you will notice the 'Downloaded' size resets to 0.0 KB

![screen shot 2017-03-06 at 3 29 05 pm](https://cloud.githubusercontent.com/assets/1534723/23628238/b56efd30-0281-11e7-8116-baff5fae1310.png)

**Escenario C - To reproduce on Mac and Windows how 'Downloaded Size' of the Files exceed the 'Size' of the file**:

1) Make sure the Status on Settings is 'Not Seeding' (Check the bottom bar)

2) Start downloading a file from YT (audio or video)

3) When the download is complete and the file changes to 'Finished' you will notice the 'Downloaded' size is way more higher than the 'real Size' of the file:

![screen shot 2017-03-06 at 3 27 06 pm](https://cloud.githubusercontent.com/assets/1534723/23628153/6c4c44f0-0281-11e7-8e2b-c817cdb06b89.png)

**Comment by gubatron** [comment]
This might also be related to several complaints related to download issues.
hopefully the changes in 1.2.0.6 might help.

**Comment by gubatron** [comment]
I think it's working fine in windows with 1.2.0.6 final.
Here's a screenshot after restarting the app
<img width="581" alt="screen shot 2017-03-06 at 6 24 03 pm" src="https://cloud.githubusercontent.com/assets/163977/23637661/2f24d7e0-029a-11e7-9b74-e380b4e7ed54.png">

**Comment by gubatron** [comment]
Waiting for confirmation that this is with 6.4.6 build 223, or slightly earlier
http://forum.frostwire.com/viewtopic.php?f=1&t=24848&p=64193#p64192

This will require some of your attention @aldenml

**Comment by gubatron** [comment]
Confirmed issues:

- There's still an issue with downloaded sizes on this 223 build (with latest jlibtorrent 1.2.0.6)
- It seems FrostWire or Jlibtorrent is not taking into account the settings of max_active_downloads and max_active_seeds. The user has max_active_downloads set to 100, but it's always limited to 12 active downloads only. And with seeds, he has 10 seeds, but he only sees 5 active seeds.

@muckachina Please try a test where you try to download 20 torrents, set the max active downloads to 100 and see if it downloads the 20 (expected), or only 12 (like the user's bug report)

**Comment by aldenml** [comment]
I'm starting to look at it

**Comment by muckachina** [comment]
Updating issue after testing this installer on Windows: (OJO this is for new downloads from scratch, old downloads or files 'downloading' before in Transfer keep getting the prior problems so the user reporting from the Forum should check only with new downloads). I sent a message via Alejandro for this user but you answered him first... 

**Escenario A - On Windows - Partially Fixed** (Still testing with new torrents)

1) Download a file and make sure the status says 'Finished' or 'Seeding'
2) Exit FrostWire and launch the app again
3) All files on the Transfer list matches the 'Size' column with the 'Downloaded' column (or vary just a bit) 
**HOWEVER**:  If the status changed to not Seeding Files appear as 'Finished and 'Downloaded' size resets to 0.0 KB

![screen shot 2017-03-07 at 9 58 29 am](https://cloud.githubusercontent.com/assets/1534723/23662178/b4729a0e-031c-11e7-82e5-3c8a87c21a6b.png)

**Escenario B - Tested only on Windows affects Cloud files** --> This issue persists

1) Make sure the Status on Settings is 'Seeding' (Check the bottom bar)
2) Start downloading a file from YT (audio or Video)
3) When the download is complete and the file changes to 'Seeding' you will notice the 'Downloaded' size resets to 0.0 KB

**Escenario C - Tested on Windows - seems FIXED!** :)

1) Make sure the Status on Settings is 'Not Seeding' (Check the bottom bar)
2) Start downloading a file from YT (audio or video)
3) When the download is complete and the file changes to 'Finished' you will notice the 'Downloaded' size is the same as the 'real Size' of the file

**Pausing files - Max number of Downloads**: (Still testing...)

This is an unstable behavior, I can download about 15 files at the same time and other 5 goes always to Pause although I Resume them again... however this may vary, it keeps downloading all files for about a minute and then some Pause. I tried what you said @gubatron with 20 files and Max downloads 100 and I could start downloading 15 without Pausing no 12... the Max number of Seeds on Settings is 10 by default (didn't change that) but still testing...

![screen shot 2017-03-07 at 9 52 30 am](https://cloud.githubusercontent.com/assets/1534723/23661888/e8691ae6-031b-11e7-8322-6cdc5902437b.png)

**Comment by muckachina** [comment]
I open a separate issue for the 'Max active Downloads' @aldenml I can reproduce it with a max of 15 Downloads not 12 as it happens at the user device, so still no consistency but definitely here only 15 files remain downloading simultaneously on both Mac and Windows.

Here the QA reference https://github.com/frostwire/qa-desktop/issues/106  Let me know If you want me to open on public repo too.

**Comment by aldenml** [comment]
@muckachina when you have a chance, test this from `master`, thanks

**Comment by muckachina** [comment]
pulling now

**Comment by muckachina** [comment]
No changes on Windows @aldenml max 15 files downloading simultaneously even after re-starting the app to double check changes on Settings. I'll pull now on Mac.

**Comment by muckachina** [comment]
Now Yes :) 

**Escenario B is fixed too!**
Now files from YT sources keep the 'Downloaded' Size no matter the Status of Seeding or not. I cannot test from SC, not getting results :( 

I still have to test on Mac but I'll close this issue unless I find something related to the 'Downloading' Size.

**Comment by muckachina** [comment]
@alejandroarturom reported the issue again with a file from YT showing a big different size after download (bigger size downloaded).

I could reproduce the problem with **big video files .MP4 on FrostWire 6.4.8 (build 230)** but in this case the 'Downloaded' size is half less than the original size indicated to download the file:

![screen shot 2017-04-18 at 10 37 15 am](https://cloud.githubusercontent.com/assets/1534723/25136755/dcc60d2e-2423-11e7-9284-ac66a8ae9549.png)


It doesn't happen with .webm or other file types from YT, only .mp4 files and only with the 'Seeding' option is not active.

**To reproduce:**

> 1) Make sure the Seeding status is 'No Seeding'
> 2) Search and download a video file from YT with Extension .MP4 
> 3) Check and compare the original size of the file with the 'Downloaded' size and you should see the difference (the bigger the size the bigger the difference)

**Comment by aldenml** [comment]
this happens to be very difficult to tackle, since there is no realistic way to know what's the actual bitrate with the crawl alone
